### ISSUE TITLE
Potential Typo gdrive

### ISSUE THREAD

**Issue by dbccomm** [watchfulli/XCloner-Wordpress#288]
within the function 

```
    public function gdrive_refresh_token($client)
    {

        $auth_token = $this->xcloner_settings->get_xcloner_option("xcloner_gdrive_access_token");
        $refresh_token = $this->xcloner_settings->get_xcloner_option("xcloner_gdrive_refresh_token");

        if (!$refresh_token) {
            $refresh_token = $auth_token['refreh_token'];
        }

        if (!$this->xcloner_settings->get_xcloner_option("xcloner_gdrive_client_id")) {
            $auth_token = $this->gdrive_app_fetch_access_token($auth_token, 'gdrive_auth_refresh');
        } else {
            $auth_token = $client->refreshToken($refresh_token);

            if ($auth_token['access_token']) {
                update_option("xcloner_gdrive_access_token", $auth_token);
                update_option("xcloner_gdrive_refresh_token", $auth_token['refresh_token']);
            } else {
                $this->xcloner->trigger_message(
                    "%s connection error: Failed to REFRESH the AUTH code - " . $auth_token['error_description'],
                    "error"
                );
            }
        }

        return $auth_token;
    }
```

I read **refreh_token** instead of refresh_token ?

**Comment by dbccomm** [comment]
I've recently encountered an issue with Google Drive integration and wanted to share my findings, hoping to get some insights or assistance.

While working with the authentication token, I found a typo in $auth_token['refreh_token'] which I corrected to $auth_token['refresh_token']. This led me to realize that $auth_token was a JSON string, so I decoded it into an array to access the 'refresh_token' attribute.

After making these changes, the connection to Google Drive began resulting in a 500 Internal Server Error when attempting to make a POST request to https://oauth.xcloner.com/google-drive/. Here's the specific error message I received:

**Server error: `POST https://oauth.xcloner.com/google-drive/` resulted in a `500 Internal Server Error` response.**
Interestingly, the process works fine as long as the token is not expired.

For context, I'm utilizing the "Sign in with Google" method, not the manual integration.

I would greatly appreciate any insights, suggestions, or assistance in resolving this issue. If more details are required, please let me know.

Thank you for your support!

**Comment by Ars92** [comment]
@dbccomm Thanks for the reporting.
We just released a new version `4.7.0` that should fix the issue.
